src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -733,32 +733,27 @@ class EthService(
         s"""Maximum number of blocks to search is $accountTransactionsMaxBlocks, requested: $numBlocksToSearch.
            |See: 'network.rpc.account-transactions-max-blocks' config.""".stripMargin)))
     } else {
-      getTransactionsFromPool map { case PendingTransactionsResponse(pendingTransactions) =>
-        val pendingSent = pendingTransactions
-          .map(_.stx)
-          .filter(_.senderAddress == request.address)
-          .map(TransactionResponse(_, pending = Some(true)))
 
-        val pendingReceived = pendingTransactions
+      def collectTxs(blockHeader: Option[BlockHeader], pending: Boolean): PartialFunction[SignedTransaction, TransactionResponse] = {
+        case stx if stx.senderAddress == request.address =>
+          TransactionResponse(stx, blockHeader, pending = Some(pending), isOutgoing = Some(true))
+        case stx if stx.tx.receivingAddress.contains(request.address) =>
+          TransactionResponse(stx, blockHeader, pending = Some(pending), isOutgoing = Some(false))
+      }
+
+      getTransactionsFromPool map { case PendingTransactionsResponse(pendingTransactions) =>
+        val pendingTxs = pendingTransactions
           .map(_.stx)
-          .filter(_.tx.receivingAddress.contains(request.address))
-          .map(TransactionResponse(_, pending = Some(true)))
+          .collect(collectTxs(None, pending = true))
 
-        val txsFromBlocks = (request.fromBlock to request.toBlock)
+        val txsFromBlocks = (request.toBlock to request.fromBlock by -1)
           .toStream
           .flatMap { n => blockchain.getBlockByNumber(n) }
           .map { block =>
```

Comment:
Minor/style: Instead `map` and latter `flatten` you can use `flatMap`

Reply:
done